### PR TITLE
Fix E2E Tests: Update ibctest (Cosmos SDK 0.46)

### LINKED ISSUE #1870: e2e tests are broken on main
## Summary

With the addition of the 46 SDK bump, the e2e tests have started failing. We should fix these

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR updates the to the version of ibctest which uses the cosmos sdk 0.46, this should allow us to run E2E tests again.

closes #1870

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -309,16 +303,6 @@ func (s *E2ETestSuite) createCosmosChains(chainOptions testconfig.ChainOptions)
...
-	s.T().Cleanup(func() {
-		if !s.T().Failed() {
-			for _, c := range []*cosmos.CosmosChain{chainA, chainB} {
-				if err := c.Cleanup(ctx); err != nil {
-					s.T().Logf("Chain cleanup for %s failed: %v", c.Config().ChainID, err)
-				}
-			}
-		}
-	})
-
```
Comment (chatton) [inline e2e/testsuite/testsuite.go:321]: the cleanup logic should be handled in ibctest now

File: e2e/go.mod [hunk 2]
```diff
@@ -139,4 +165,8 @@ require (
 	modernc.org/sqlite v1.17.3 // indirect
 	modernc.org/strutil v1.1.1 // indirect
 	modernc.org/token v1.0.0 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+// TODO: using version v1.0.0 causes a build failure. This is the previous version which compiles successfully.
+replace github.com/ChainSafe/go-schnorrkel => github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply e2e/go.mod:172]: weird

**Review by damiannolan — APPROVED** [review]
Cheers 🥂

**Review by colin-axner — COMMENTED** [review]
post merge ACK

**Pull request merged**